src/pcp/types.rs

```diff
@@ -143,11 +161,19 @@ impl<F: FieldElement> Value<F, PolyEval<F>> for PolyCheckedVector<F> {
         &self.data
     }
 
-    fn new_with(&self, data: Vec<F>) -> Self {
-        Self {
-            data,
-            poly: self.poly.clone(),
-        }
+    fn param(&self) -> Self::Param {
+        self.poly.clone()
+    }
+}
+
+impl<F: FieldElement> TryFrom<(Vec<F>, Vec<F>)> for PolyCheckedVector<F> {
```

Comment:
It seems like it'd be easy to accidentally confuse the vector of `FieldElement` and the `Vec<F>` that's actually a polynomial. Absent a real `Polynomial` type, could this be written as `TryFrom<(Vec<F>, PolyCheckedVector<F>::Param)>` (not sure if that compiles, it's just a guess)?

Reply:
Hmmm, I can't seem to spell this in a way that compiles.